src/examples/java/io/nats/examples/jetstream/simple/FetchExample.java

```diff
@@ -18,103 +18,150 @@
 
             // 1. Different fetch max messages demonstrate expiration behavior
 
             // 1A. equal number of messages to the fetch max messages
-            simpleFetch("1A", nc, 20, 0);
+            simpleFetch(nc, js, "1A", 20, 0);
 
             // 1B. more messages than the fetch max messages
-            simpleFetch("1B", nc, 10, 0);
+            simpleFetch(nc, js, "1B", 10, 0);
 
             // 1C. fewer messages than the fetch max messages
-            simpleFetch("1C", nc, 40, 0);
+            simpleFetch(nc, js, "1C", 40, 0);
+
+            // 1D. "fetch-consumer-40-messages" was created in 1C and has no messages available
+            simpleFetch(nc, js, "1D", 40, 0);
+
+            // bytes don't work before server v2.9.1
+            if (nc.getServerInfo().isOlderThanVersion("2.9.1")) {
+                return;
+            }
 
             // 2. Different max bytes sizes demonstrate expiration behavior
             //    - each test message is approximately 100 bytes
 
             // 2A. max bytes is reached before message count
-            simpleFetch("2A", nc, 10, 750);
+            simpleFetch(nc, js, "2A", 0, 700);
 
             // 2B. fetch max messages is reached before byte count
-            simpleFetch("2B", nc, 10, 1500);
+            simpleFetch(nc, js, "2B", 10, 1500);
 
             // 2C. fewer bytes than the byte count
-            simpleFetch("2C", nc, 25, 3000);
-
-            // 3. simple-consumer-40msgs was created in 1C and has no messages available
-            simpleFetch("3", nc, 40, 0);
+            simpleFetch(nc, js, "2C", 0, 3000);
+        }
+        catch (IOException ioe) {
+            // problem making the connection or
         }
-        catch (Exception e) {
-            e.printStackTrace();
+        catch (InterruptedException e) {
+            // thread interruption in the body of the example
         }
     }
 
-    private static void simpleFetch(String label, Connection nc, int maxMessages, int maxBytes) throws IOException, JetStreamApiException, InterruptedException {
-        JetStreamManagement jsm = nc.jetStreamManagement();
-        JetStream js = nc.jetStream();
-
-        String name = generateConsumerName(maxMessages, maxBytes);
+    private static void simpleFetch(Connection nc, JetStream js, String label, int maxMessages, int maxBytes) {
+        String consumerName = generateConsumerName(maxMessages, maxBytes);
 
-        // Pre define a consumer
-        ConsumerConfiguration cc = ConsumerConfiguration.builder().durable(name).build();
-        jsm.addOrUpdateConsumer(STREAM, cc);
-
-        // Consumer[Context]
-        ConsumerContext consumerContext = js.getConsumerContext(STREAM, name);
+        // get stream context, create consumer and get the consumer context
+        StreamContext streamContext;
+        ConsumerContext consumerContext;
+        try {
+            streamContext = nc.streamContext(STREAM);
```

Comment:
There actually is a `getStreamContext` and `getConsumerContext` off of `js`. I probably should make those consistent (remove the `get`).
Hopefully I'll get more feedback here.

Reply:
To me it would sound logical to have to convert the `nc: NatsConnection` to `nc.jetStream()` first, so you are explicitly in the JetStream context. And only then having access to the `js.getStreamContext / js.getConsumerContext`.

That is, if you'd still want to make the distinction though. Since it would mean that the simplification is two calls away instead of just one (`nc.jetStream().getStreamContext()` vs `nc.streamContext()`). I would consider having it accessible more quickly to be valuable as well.

Reply:
I would probably lean towards just having `js.getStreamContext()` (or `js.streamContext()`) instead of having those methods on both `NatsConnection` and `JetStream`. Having it one additional call away does not really make it worse IMO and gives you clean separation - `NatsConnction` should not be handling streams, consumers etc.